File: 1118.py
cont=0
media=0
while True:
    try:
        nota=float(input())
        if nota>=0 and nota<=10:
            media+=nota
            cont+=1
        if nota<0 or nota>10:
            print('nota invalida')
            pass
        if cont==2:
            media=float(media/2)
            print('media = %.2f'%media)
            print('novo calculo (1-sim 2-nao)')
            resposta=int(input())
            if respota==1:
                cont=0
                media=0
            elif resposta==2:
                break
            elif resposta>1 and resposta!=2 or resposta<1 and resposta!=2:
                print('novo calculo (1-sim 2-nao)')
                resposta = int(input())
                if resposta==2:
                    break
                else:
                    cont=0
                    media=0
    except:
        EOFError

File: C/1118.c
#include <stdio.h>
#include <stdlib.h>

void main(){
    int cont=0,resposta;
    double media=0,nota1,nota2;
    while(cont<2){
        scanf("%lf",&nota1);
        scanf("%lf",&nota2);
        if(nota1>=0&&nota1<=10){
            media=media+nota1;
            cont=cont+1;
        }
        if(nota2>=0&&nota2<=10){
            media=media+nota2;
            cont=cont+1;
        }
        if(nota1<0||nota1>10){
            printf("nota invalida\n");
        }
        if(nota2<0||nota2>10){
            printf("nota invalida\n");
        }
        if(cont==2){
            media=media/2.00;
            printf("media = %.2lf\n",media);
            printf("novo calculo (1-sim 2-nao)\n");
            scanf("%d",&resposta);
            if(resposta==1){
                cont=0;
                media=0;
            }
            if(resposta==2){
                break;           
            }
            if(resposta>1||resposta<1&&resposta!=2){
                cont=1;
                printf("novo calculo (1-sim 2-nao)\n");
                scanf("%d",&resposta);
                if(resposta==2){
                    break;
                }
                else{
                    cont=0;
                    media=0;
                }
            }
        }
    }
}